test/e2e/e2e_suite_test.go

```diff
@@ -208,6 +214,38 @@ func initBootstrapCluster(bootstrapClusterProxy framework.ClusterProxy, config *
 	}, config.GetIntervals(bootstrapClusterProxy.GetName(), "wait-controllers")...)
 }
 
+func dumpBootstrapClusterLogs(bootstrapClusterProxy framework.ClusterProxy) {
+	if bootstrapClusterProxy == nil {
+		return
+	}
+
+	clusterLogCollector := bootstrapClusterProxy.GetClusterLogCollector()
+	if clusterLogCollector == nil {
+		return
+	}
+
+	nodes, err := bootstrapClusterProxy.GetClientSet().CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("Failed to get nodes for the bootstrap cluster: %v\n", err)
+		return
+	}
+
+	// We assume that the test framework generated bootstrap cluster has only one node.
+	bootstrapNodeName := nodes.Items[0].GetName()
+	err = clusterLogCollector.CollectMachineLog(
+		context.TODO(),
+		bootstrapClusterProxy.GetClient(),
+		&clusterv1.Machine{
+			Spec:       clusterv1.MachineSpec{ClusterName: bootstrapNodeName},
+			ObjectMeta: metav1.ObjectMeta{Name: bootstrapNodeName},
+		},
```

Comment:
What about adding a note here or just before `clusterLogCollector.CollectMachineLog`
// The bootstrap cluster is not expected to be a CAPI cluster, so in order to re-use the logCollector, we create a fake machine that wraps the node.
// NOTE: this assumes a naming convention between machines and nodes, which e.g. applies to the bootstrap clusters generated with kind; this might not work in case using an existing bootstrap cluster provided by other means.

Reply:
Done!